plugin/cache.go

```diff
@@ -189,7 +190,21 @@ func cachedList(ctx context.Context, p Parent) (map[string]Entry, error) {
 		return nil, err
 	}
 
-	return cachedEntries.(map[string]Entry), nil
+	// Return a copy of the map so we don't mutate the cached version. Future calls to list should
```

Comment:
If we're going to use an attribute on the entry to determine whether we have permission to operate on it, deciding whether or not to include that entry in a list of entries can be moved closer to where it's displayed to the user.